pkg/webhook/defaults/pods.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020-2021 Clastix Labs
+// SPDX-License-Identifier: Apache-2.0
+
+package defaults
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	schedulev1 "k8s.io/api/scheduling/v1"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	capsulev1beta2 "github.com/clastix/capsule/api/v1beta2"
+	"github.com/clastix/capsule/pkg/webhook/utils"
+)
+
+func mutatePodDefaults(ctx context.Context, req admission.Request, c client.Client, decoder *admission.Decoder, recorder record.EventRecorder) *admission.Response {
+	var err error
+
+	pod := &corev1.Pod{}
+	if err = decoder.Decode(req, pod); err != nil {
+		response := admission.Errored(http.StatusInternalServerError, err)
+
+		return &response
+	}
+
+	var tnt *capsulev1beta2.Tenant
+
+	tnt, err = utils.TenantByStatusNamespace(ctx, c, pod.Namespace)
+	if err != nil {
+		return utils.ErroredResponse(err)
+	}
+
+	if tnt == nil {
+		return nil
+	}
+
+	allowed := tnt.Spec.PriorityClasses
+
+	if allowed == nil || allowed.Default == "" {
+		return nil
+	}
+
+	priorityClassPod := pod.Spec.PriorityClassName
+	mutate := false
+
+	var cpc *schedulev1.PriorityClass
+
+	// PriorityClass name is empty, if no GlobalDefault is set and no PriorityClass was given on pod
+	if priorityClassPod != "" {
+		if priorityClassPod != allowed.Default {
+			cpc, err = utils.GetPriorityClassByName(ctx, c, priorityClassPod)
+			// Should not happen, since API already checks if PC present
+			if err != nil {
+				response := admission.Denied(NewPriorityClassError(priorityClassPod, err).Error())
+
+				return &response
+			}
+		}
+	} else {
```

Comment:
```suggestion
	if priorityClassPod != "" && priorityClassPod != allowed.Default {
		cpc, err = utils.GetPriorityClassByName(ctx, c, priorityClassPod)
		// Should not happen, since API already checks if PC present
		if err != nil {
			response := admission.Denied(NewPriorityClassError(priorityClassPod, err).Error())

			return &response
		}
	} else {
```